lib/stan/fit_result.rb

```diff
@@ -3,9 +3,35 @@ class FitResult
     attr_reader :rows, :headers, :data_rows
 
     def initialize(original_output)
-      @rows = Array(original_output.split("\n"))
-      @headers = rows[38].to_s.split(",") # In L39 there is the header column
-      @data_rows = rows[43..].map { |r| r.split(",") } # From L44 on there is the raw data!
+      # sometimes \r remains without the chomp
+      @rows = Array(original_output.split("\n").map(&:chomp))
+
+      # set headers
+      headers_index = rows.index do |row|
+        header_row?(row)
+      end
+      @headers = rows[headers_index].to_s.split(",")
+
+      # set data rows
+      data_rows_start_index = nil
+      (headers_index + 1).upto(rows.length - 1) do |row_index|
+      row = rows[row_index]
+        if data_row?(row)
+          data_rows_start_index = row_index
+          break
+        end
+      end
+
+      # last few lines are not actually data values but elapsed time
+      data_rows_end_index = nil
+      (rows.length - 1).downto(headers_index + 1) do |row_index|
+        row = rows[row_index]
+        if data_row?(row)
+          data_rows_end_index = row_index
+          break
+        end
```

Comment:
I think you can simplify this a little bit. Maybe `Enumberable#detect` is a nice improvement for your task. Given a list to iterate ver, it returns the first item matching a certain condition. Like this:

```Ruby
data_rows_end_index = 
  (rows.length - 1).downto(headers_index + 1).detect do |row_index|
    data_row?(rows[row_index])
  end
```
What do you think?  

Your solution is technically fine and we can roll with it if I got something wrong :) But it's fiun boiling things down, isn't it? :)

Reply:
Yes! That's exactly what I want.
I am still a beginner at Ruby, so I did not have `detect` in my mind.
Thank you for improving both my code and me!